cmd/snap/cmd_connections.go

```diff
@@ -0,0 +1,209 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2018 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package main
+
+import (
+	"fmt"
+	"sort"
+	"strings"
+
+	"github.com/jessevdk/go-flags"
+
+	"github.com/snapcore/snapd/client"
+	"github.com/snapcore/snapd/i18n"
+)
+
+type cmdConnections struct {
+	clientMixin
+	All         bool `long:"all"`
+	Positionals struct {
+		Snap installedSnapName
+	} `positional-args:"true"`
+}
+
+var shortConnectionsHelp = i18n.G("List interface connections")
+var longConnectionsHelp = i18n.G(`
+The connections command lists connections between plugs and slots
+in the system.
+
+Pass --all to list unconnected plugs and slots alongside those
+already connected. Unless <snap> is provided, the listing is for all
+snaps in the system.
+
+$ snap connections <snap>
+
+Lists connected and unconnected plugs and slots for the specified
+snap.
+`)
+
+func init() {
+	addCommand("connections", shortConnectionsHelp, longConnectionsHelp, func() flags.Commander {
+		return &cmdConnections{}
+	}, map[string]string{
+		"all": i18n.G("Show connected and unconnected plugs and slots"),
+	}, []argDesc{{
+		// TRANSLATORS: This needs to be wrapped in <>s.
+		name: "<snap>",
+		// TRANSLATORS: This should not start with a lowercase letter.
+		desc: i18n.G("Constrain listing to a specific snap"),
+	}})
+}
+
+func isSystemSnap(snap string) bool {
+	return snap == "core" || snap == "snapd" || snap == "system"
+}
+
+func endpoint(snap, name string) string {
+	if isSystemSnap(snap) {
+		return ":" + name
+	}
+	return snap + ":" + name
+}
+
+func wantedSnapMatches(name, wanted string) bool {
```

Comment:
this doesn't seem to be used, is it needed? missing tests?

Reply:
`unused` (from `go get honnef.co/go/tools/cmd/unused`) says no:
```
$ unused ./cmd/snap
Unused has been deprecated. Please use staticcheck instead.
cmd/snap/cmd_advise.go:114:3: field Command is unused (U1000)
cmd/snap/cmd_advise.go:115:3: field SearchTerms is unused (U1000)
cmd/snap/cmd_connections.go:80:6: func wantedSnapMatches is unused (U1000)
cmd/snap/cmd_connections.go:112:6: func connName is unused (U1000)
cmd/snap/cmd_connectivity_check.go:47:3: field Connectivity is unused (U1000)
cmd/snap/cmd_snapshot.go:33:6: func fmtSize is unused (U1000)
cmd/snap/cmd_snapshot_test.go:32:6: type snapshotCmdArgs is unused (U1000)
```